### PR TITLE
[MINOR] Censor sensitive fields in the request and response log

### PR DESCRIPTION
**Pull request by nickwilliams-eventbrite**

### CODE REVIEW

File: pysoa/server/logging.py [hunk 1]
```diff
@@ -36,3 +38,151 @@ def get_logging_request_context(cls):
...
+    # 10000000 loops, best of 3: 131 ns per loop
+    #
+    # In [3]: %timeit random_string()
+    # 100000 loops, best of 3: 10.4 us per loop
+    #
+    # In [4]: %timeit random_string() in SENSITIVE_FIELDS
+    # 100000 loops, best of 3: 10.7 us per loop
+    #
+    # Suffice it to say, a lookup against this set should always take less than 250 ns.
+    SENSITIVE_FIELDS = frozenset({
```
Comment (adamsussman) [inline pysoa/server/logging.py]: I would add the auth token to this list.  Encrypted body content in general ought to be blanked.

Comment (adamsussman) [inline pysoa/server/logging.py]: How does a given service customize this list for itself?

Comment (nickwilliams-eventbrite) [reply pysoa/server/logging.py]: We discussed that in the squad channel on Slack. A request was made to //temporarily// exclude auth tokens from this while we work work out the kinks in our new internal authentication service, and then eventually add it to this list.

File: pysoa/server/logging.py [hunk 2]
```diff
@@ -36,3 +38,151 @@ def get_logging_request_context(cls):
...
+    def __bytes__(self):
+        # If this method is called, we must be in Python 3, which means __str__ must be returning a string
+        return self.__str__().encode('utf-8')
+
+    def __unicode__(self):
+        # If this method is called, we must be in Python 2, which means __str__ must be returning bytes
+        return self.__str__().decode('utf-8')
+
+    @classmethod
+    def _copy_and_censor_unknown_value(cls, v, should_censor_strings):
```
Comment (vartec) [inline pysoa/server/logging.py]: at the point where it's used, can the value be any custom type?

Comment (nickwilliams-eventbrite) [reply pysoa/server/logging.py]: I'm not sure what you mean by this. Can you re-phrase?

Comment (vartec) [reply pysoa/server/logging.py]: can `v` be anything but the basic types?

File: pysoa/server/logging.py (hunk 1 above)
Comment (andrewgodwin) [reply pysoa/server/logging.py]: I'm not sure that will work forever, but this is a decent enough start.

File: pysoa/server/logging.py [hunk 3]
```diff
@@ -36,3 +38,151 @@ def get_logging_request_context(cls):
...
+        if isinstance(v, dict):
+            return cls._copy_and_censor_dict(v)
+
+        if isinstance(v, (list, set, tuple)):
```
Comment (andrewgodwin) [inline pysoa/server/logging.py]: I'd add `frozenset` to this list.

File: pysoa/server/logging.py (hunk 2 above)
Comment (nickwilliams-eventbrite) [reply pysoa/server/logging.py]: Yes. If it's not one of the basic types, it won't be censored, but we only support serialization/deserialization of the basic types, anyway, so this isn't a problem.

File: pysoa/server/logging.py (hunk 3 above)
Comment (nickwilliams-eventbrite) [reply pysoa/server/logging.py]: Oh, good catch. I had thought `frozenset` was-a `set`. Turns out it's not.

**Pull request merged**